packages/alea-frontend/components/ExamSchedule.tsx

```diff
@@ -0,0 +1,64 @@
+import { Box, Typography } from '@mui/material';
+import QuizIcon from '@mui/icons-material/Quiz';
+
+export default function ExamSchedule({ examDates }) {
+  if (!examDates?.length) return null;
+
+  const content = examDates.map((exam, idx) => {
+    const hasTime = exam.examStartTime && exam.examEndTime;
+
+    const start = new Date(`${exam.examDate}T${exam.examStartTime || '00:00'}:00+02:00`);
+    const end = new Date(`${exam.examDate}T${exam.examEndTime || '00:00'}:00+02:00`);
+
+    const formattedDate = start.toLocaleDateString('en-GB', {
+      day: '2-digit',
+      month: 'short',
+      year: 'numeric',
+      timeZone: 'Europe/Berlin',
+    });
+
+    const formattedStart = hasTime
+      ? start.toLocaleTimeString('en-GB', {
+          hour: '2-digit',
+          minute: '2-digit',
+          timeZone: 'Europe/Berlin',
+        })
+      : '';
+    const formattedEnd = hasTime
+      ? end.toLocaleTimeString('en-GB', {
+          hour: '2-digit',
+          minute: '2-digit',
+          timeZone: 'Europe/Berlin',
+        })
+      : '';
+
+    return (
+      <Typography key={idx} variant="body2" sx={{ color: '#e65100', mb: 1, fontWeight: 500 }}>
+        🗓 {formattedDate}
+        {hasTime && `   ⏰ ${formattedStart} – ${formattedEnd} (Europe/Berlin)`}
+      </Typography>
+    );
+  });
+
+  return (
+    <Box
+      sx={{
+        px: { xs: 1, sm: 2 },
+        py: { xs: 1, sm: 1.5 },
+        mt: 2,
+        borderRadius: '8px',
+        background: 'linear-gradient(135deg, #fff3e0, #ffe0b2)',
+        border: '1px solid #ffcc80',
+      }}
+    >
+      <Box sx={{ display: 'flex', alignItems: 'center', gap: 1.5, mb: 1.5 }}>
+        <QuizIcon sx={{ color: '#bf360c', fontSize: '20px' }} />
+        <Typography variant="h6" sx={{ fontWeight: 600, color: '#bf360c', fontSize: '1rem' }}>
+          Exam Schedule
+        </Typography>
+      </Box>
+
+      {content}

```

Comment:
This is not wrong but somewhat unusual organization of jsx. typically, you would put a loop in the main jsx itself (typically we say "Your code is not idiomatic"). The return statement of a component should, as much as possible, describe the entire rendered output. When you define JSX in variables beforehand, a developer has to look in multiple places to understand the component's final structure.

Instead, You could consider creating a separate component (perhaps in the same file) to make things more readable (and reusable)
```javascript
export default function ExamDate({ exam... })
  const hasTime = ...;

  const start = ...
  const end = ...;
```


```suggestion
    {examDates.map((exam, idx) => (
        <ExamDate key={idx} exam={exam} />
      ))}
```